### ISSUE TITLE
[chip_if] tap_strap_if.pins as high-Z makes IOs unknown.

### ISSUE THREAD

**Issue by eunchan** [lowRISC/opentitan#15006]
chip_if adds muxes to MIOs for JTAG interface. `__enable_jtag` becomes high-Z when `tap_straps_if` is disconnected. 

<img width="1804" alt="image" src="https://user-images.githubusercontent.com/1192814/191109506-da4fdcd0-004a-4a38-aaad-3fa42c35f7da.png">

The tap_strap_if pins are also connected to IOC5, IOC8 MIOs.

**Comment by tjaychen** [comment]
yeah this one is a bit problematic. 
we can't really rely on the pins directly for tap strap validity. It should be a combination of life cycle states, only tap-strap0 is more or less dedicated in function.

**Comment by eunchan** [comment]
> yeah this one is a bit problematic. we can't really rely on the pins directly for tap strap validity. It should be a combination of life cycle states, only tap-strap0 is more or less dedicated in function.
>
Yup... I could not think of nice solution here as I do not know the details of the chip interface. So I just skip those pins from testing candidate.

**Comment by weicaiyang** [comment]
should we just decouple `__enable_jtag` with `tap_straps_if.pins`, and let user decide to connect the jtag or not, like this?

```
  bit enable_jtag;

  assign ios[IoR0] = enable_jtag ? jtag_if.tms : 1'bz;
  assign jtag_if.tdo = enable_jtag ? ios[IoR1] : 1'bz;
  assign ios[IoR2] = enable_jtag ? jtag_if.tdi : 1'bz;
  assign ios[IoR3] = enable_jtag ? jtag_if.tck : 1'bz;
  assign ios[IoR4] = enable_jtag ? jtag_if.trst_n : 1'bz;
```

**Comment by eunchan** [comment]
So, the assumption is, always TB decides the JTAG connection (by driving IOC5, IOC8)? Then adding bit seems to be good.

**Comment by weicaiyang** [comment]
> So, the assumption is, always TB decides the JTAG connection (by driving IOC5, IOC8)? Then adding bit seems to be good.

yes, TB needs to decide that.

**Comment by sriyerg** [comment]
> should we just decouple `__enable_jtag` with `tap_straps_if.pins`, and let user decide to connect the jtag or not, like this?
> 
> ```
>   bit enable_jtag;
> 
>   assign ios[IoR0] = enable_jtag ? jtag_if.tms : 1'bz;
>   assign jtag_if.tdo = enable_jtag ? ios[IoR1] : 1'bz;
>   assign ios[IoR2] = enable_jtag ? jtag_if.tdi : 1'bz;
>   assign ios[IoR3] = enable_jtag ? jtag_if.tck : 1'bz;
>   assign ios[IoR4] = enable_jtag ? jtag_if.trst_n : 1'bz;
> ```

IMO setting the TAP straps should be enough intent to enable JTAG. I have fixed this in #15017.
